src/design.jl

```diff
@@ -97,6 +97,40 @@ end
 """
 $(TYPEDSIGNATURES)
 
+```jldoctest
+julia> fold!(PlackettBurman(2))
+8×3 DataFrame
+│ Row │ factor1 │ factor2 │ dummy1 │
+│     │ Int64   │ Int64   │ Int64  │
+├─────┼─────────┼─────────┼────────┤
+│ 1   │ 1       │ 1       │ 1      │
+│ 2   │ 1       │ -1      │ -1     │
+│ 3   │ -1      │ -1      │ 1      │
+│ 4   │ -1      │ 1       │ -1     │
+│ 5   │ -1      │ -1      │ -1     │
+│ 6   │ -1      │ 1       │ 1      │
+│ 7   │ 1       │ 1       │ -1     │
+│ 8   │ 1       │ -1      │ 1      │
+
+```
+"""
+function fold!(design::AbstractDesign)
```

Comment:
I think the best solution would be to implement a `mirror!` method for `PlackettBurman` designs like you did before, and then adding specific methods of `mirror!` for different designs.

Reply:
The thing is, I was working already on 2-level fractional factorial designs (https://github.com/joergbuchwald/ExperimentalDesign.jl/tree/fractfact) and just saw that the same function could be used for fractional factorial design as well. 
My suggestion would be to construct an abstract type that contains PlackettBurman as well as fractional factorial designs. What do you think?

Reply:
That's broader, but it would still be applicable only to 2-level designs with the current function.

I would prefer using multiple dispatch here to target different design types.

A "2-level abstract" type would also fix this, but I don't think it would be a good idea, because we can have fractional factorial designs with more than 2 levels, and they would be of different types.

Reply:
We could also have a "fold.jl" file, with all implementations for different types.

Reply:
I think it is not wrong to call both screening designs.

Reply:
That's a good idea, I think

Reply:
From an implementation point of view, how do you suggest we treat 2- and 3-level designs differently?

Reply:
Of course, there could be a fractional factorial design with more than two levels, however, they are not very common.
(E.g. wikipedia states "In practice, one rarely encounters l > 2 levels in fractional factorial designs, since response surface methodology is a much more experimentally efficient way to determine the relationship between the experimental response and factors at multiple levels. In addition, the methodology to generate such designs for more than two levels is much more cumbersome.")
So thinking about implementation, I'm not sure whether it is worth to think about multilevel screening designs right now.
So if you think it makes sense to consider them in the future, it would be enough to create some abstract types for now.

Reply:
That's a good plan. We can add a type for multi-level screening designs later if we need it.